pkg/controllers/restore/controller.go

```diff
@@ -370,6 +370,12 @@ func (h *handler) generateDependencyGraph(ownerToDependentsList map[string][]res
 		}
 
 		customize(&resourceData)
+		if resourceData.GetKind() == "Deployment" && namespace == "cattle-system" {
+			if strings.HasSuffix(name, "rancher") || strings.HasSuffix(name, "rancher-webhook") {
```

Comment:
Are there cases where Rancher / webhook deployments are named anything different than `rancher` and `rancher-webhook` exactly? If not, we don't need `HasSuffix` checks.

Reply:
Currently, we have only those 2. 
I added the check to avoid deleting others if we have them in the future. 

Should I  not worry about it now?

Reply:
`name == "rancher"` is more restrictive than `strings.HasSuffix(name, "rancher")` - "rancher-something" is false for the former and true for the latter.
So, if we want to avoid deleting others - we should go with more restrictive check.

I was wondering if you went with `HasSuffix` because you know of deployments named like `rancher-something` that should also be skipped. Looks like it's not the case.

Reply:
If the user sets a different name when installing rancher, for example, `helm install my-setup rancher-latest/rancher`, the deployment’s name will be `my-setup-rancher`. `strings.HasSuffix(name, "rancher")` is to catch this case.

Reply:
Thank you for the explanation, makes sense. Also, my example in comment was incorrect (suffix vs. prefix mix-up), I've corrected it.